huaweicloud/services/eg/resource_huaweicloud_eg_event_subscription_target.go

```diff
@@ -0,0 +1,752 @@
+		d.Set("created_at", utils.PathSearch("created_time", target, nil)),
+		d.Set("updated_at", utils.PathSearch("updated_time", target, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceEventSubscriptionTargetUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg            = meta.(*config.Config)
+		region         = cfg.GetRegion(d)
+		subscriptionId = d.Get("subscription_id").(string)
+		targetId       = d.Id()
+		httpUrl        = "v1/{project_id}/subscriptions/{subscription_id}/targets/{target_id}"
+	)
+
+	client, err := cfg.NewServiceClient("eg", region)
+	if err != nil {
+		return diag.Errorf("error creating EG client: %s", err)
+	}
+
+	updatePath := client.Endpoint + httpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updatePath = strings.ReplaceAll(updatePath, "{subscription_id}", subscriptionId)
+	updatePath = strings.ReplaceAll(updatePath, "{target_id}", targetId)
+	updatePath += buildEventSubscriptionTargetQueryParams(d)
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+		JSONBody:         buildEventSubscriptionTargetBodyParams(d),
+	}
+
+	_, err = client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return diag.Errorf("error updating EG event subscription target: %s", err)
+	}
+
+	return resourceEventSubscriptionTargetRead(ctx, d, meta)
+}
+
+func resourceEventSubscriptionTargetDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg            = meta.(*config.Config)
+		region         = cfg.GetRegion(d)
+		httpUrl        = "v1/{project_id}/subscriptions/{subscription_id}/targets/{target_id}"
+		subscriptionId = d.Get("subscription_id").(string)
+		targetId       = d.Id()
+	)
+
+	client, err := cfg.NewServiceClient("eg", region)
+	if err != nil {
+		return diag.Errorf("error creating EG client: %s", err)
+	}
+
+	deletePath := client.Endpoint + httpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = strings.ReplaceAll(deletePath, "{subscription_id}", subscriptionId)
+	deletePath = strings.ReplaceAll(deletePath, "{target_id}", targetId)
+	deleteOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	_, err = client.Request("DELETE", deletePath, &deleteOpt)
+	if err != nil {
+		return diag.Errorf("error deleting EG event subscription target: %s", err)
+	}
+
+	return nil
+}
+
+func resourceEventSubscriptionTargetImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	importedId := d.Id()
+	parts := strings.Split(importedId, "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import ID, want '<subscription_id>/<id>', but got '%s'",
+			importedId)
+	}
+
+	d.SetId(parts[1])
+	d.Set("subscription_id", parts[0])
+	return []*schema.ResourceData{d}, nil
```

Comment:
```suggestion

	return []*schema.ResourceData{d}, d.Set("subscription_id", parts[0])
```
report error if the subscription ID set failed

Reply:
done